plugins/optimization-detective/optimization.php

```diff
@@ -31,8 +31,35 @@
  * @return string Unmodified value of $passthrough.
  */
 function od_buffer_output( string $passthrough ): string {
+	/*
+	 * Instead of the default PHP_OUTPUT_HANDLER_STDFLAGS (cleanable, flushable, and removable) being used for flags,
+	 * we need to omit PHP_OUTPUT_HANDLER_FLUSHABLE. If the buffer were flushable, then each time that ob_flush() is
+	 * called, it would send a fragment of the output into the output buffer callback. When buffering the entire
+	 * response as an HTML document, this would result in broken HTML processing.
+	 *
+	 * If this ends up being problematic, then PHP_OUTPUT_HANDLER_FLUSHABLE could be added to the $flags and the
+	 * output buffer callback could check if the phase is PHP_OUTPUT_HANDLER_FLUSH and abort any subsequent
+	 * processing while also emitting a _doing_it_wrong().
+	 */
+	$flags = PHP_OUTPUT_HANDLER_CLEANABLE;
+
+	// When running unit tests the output buffer must also be removable in order to obtain the buffered output.
+	if ( php_sapi_name() === 'cli' ) {
+		// TODO: Do any caching plugins need the output buffer to be removable? This is unlikely, as they would pass an output buffer callback to ob_start() instead of calling ob_get_clean() at shutdown.
+		$flags |= PHP_OUTPUT_HANDLER_REMOVABLE;
+	}
```

Comment:
This actually does need to be removable because WordPress core calls `wp_ob_end_flush_all()` at `shutdown`, and the lack of `PHP_OUTPUT_HANDLER_REMOVABLE` results in:

> Notice: ob_end_flush(): Failed to send buffer of Closure::__invoke (1) in /var/www/html/wp-includes/functions.php on line 5427

Reply:
Addressed in 0b4f2cd6a871ab30df01152b277d27dd6803d289